dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedWorkflowResource.java

```diff
@@ -180,15 +182,43 @@ public Workflow deleteHostedVersion(User user, Long entryId, String version) {
     }
 
     @Override
-    protected boolean checkValidVersion(Set<SourceFile> sourceFiles, Workflow entry) {
+    protected WorkflowVersion versionValidation(WorkflowVersion version, Workflow entry) {
+        Set<SourceFile> sourceFiles = version.getSourceFiles();
         SourceFile.FileType identifiedType = entry.getFileType();
         String mainDescriptorPath = this.descriptorTypeToDefaultDescriptorPath.get(entry.getDescriptorType().toLowerCase());
-        for (SourceFile sourceFile : sourceFiles) {
-            if (Objects.equals(sourceFile.getPath(), mainDescriptorPath)) {
-                return LanguageHandlerFactory.getInterface(identifiedType).isValidWorkflow(sourceFile.getContent());
-            }
+        Optional<SourceFile> mainDescriptor = sourceFiles.stream().filter((sourceFile -> Objects.equals(sourceFile.getPath(), mainDescriptorPath))).findFirst();
+
+        // Validate descriptor set
+        ImmutablePair validDescriptorSet;
+        Validation descriptorValidation;
+        if (mainDescriptor.isPresent()) {
+            validDescriptorSet = LanguageHandlerFactory.getInterface(identifiedType).validateWorkflowSet(sourceFiles, mainDescriptorPath);
+        } else {
+            validDescriptorSet = new ImmutablePair(false, "Missing the primary descriptor.");
+        }
+        descriptorValidation = new Validation(identifiedType, validDescriptorSet);
+        version.addOrUpdateValidation(descriptorValidation);
+
+        // Validate test parameter set
+        SourceFile.FileType testParameterType = SourceFile.FileType.CWL_TEST_JSON;
```

Comment:
Nextflow eventually, this will default Nextflow content to CWL when it should probably trigger a fast fail or work properly.